src/strategies/Staking.sol

```diff
@@ -4,174 +4,348 @@ pragma solidity =0.8.16;
+        // Pay debt in compound
+        uint ethDebt = cETH.borrowBalanceCurrent(address(this));
+        IWETH(address(WETH)).withdraw(ethBorrowed);
+
+        cETH.repayBorrow{value: ethBorrowed}();
+        uint daiToRedeem = cDAI.balanceOfUnderlying(address(this)).mulDivDown(ethBorrowed, ethDebt);
+        uint res = cDAI.redeemUnderlying(daiToRedeem);
+        require(res == 0, "Staking: comp redeem error");
+
+        // Pay debt in maker
+
+        // Send the dai to urn
+        address urn = MAKER.urns(cdpId);
+        joinDai.join({usr: urn, wad: daiToRedeem});
+
+        // Pay debt. Collateral withdrawn proportional to debt paid
+        (uint wstEthCollat,)  = VAT.urns(ilk, urn);
+        uint wstEthToRedeem = wstEthCollat.mulDivDown(ethBorrowed, ethDebt);
+        MAKER.frob(cdpId, -int(wstEthToRedeem), -int(daiToRedeem));
+
+        // Withdraw wstEth from maker
+        MAKER.flux({cdp: cdpId, dst: address(this), wad: wstEthToRedeem});
+        WSTETH_JOIN.exit(address(this), wstEthToRedeem);
+
+        // Convert from wrapped staked eth => stEth 
+        LIDO.unwrap(wstEthToRedeem);
+
+
+        // Convert stEth => eth to pay back flashloan and pay back user
+        // Trade on stEth for ETH (steth is at index 1)
+        CURVE.exchange({x: 1, y: 0 , dx: STETH.balanceOf(address(this)), min_dy: 0}); // TODO: slippage protection when withdrawing from curve
+
+         // Convert to weth
+        IWETH(address(WETH)).deposit{value: address(this).balance}();
+
+        // Convert weth => dai to pay back flashloan
+        if (daiBorrowed > 0) {
+             ISwapRouter.ExactInputSingleParams memory paramsCrv = ISwapRouter.ExactInputSingleParams({
+                tokenIn: address(WETH),
+                tokenOut: address(DAI),
+                fee: 500,
+                recipient: address(this),
+                deadline: block.timestamp,
+                amountIn: (daiBorrowed / 1978).mulDivUp(110, 100),
+                amountOutMinimum: daiBorrowed,
+                sqrtPriceLimitX96: 0
+            });
+        }
+        console.log("balance of dai after trade: ", DAI.balanceOf(address(this)));
+       
+    }
 
-        // Payback loan
-        asset.safeTransfer(address(vault), amount);
+    // TODO: remove this and startPosition
+    function endPosition(uint _amount) external onlyRole(STRATEGIST_ROLE) {
+        _divest(_amount);
     }
 
-    function withdrawEth(uint256 amount, address to) external {
-        require(msg.sender == owner, "Staking: only owner can call");
-        (bool success,) = to.call{value: amount}("");
-        require(success, "Staking: transfer failed");
+
+    /// @notice Borrow from maker and supply to compound or vice-versa.
+    /// @param amountEth Amount of eth to borrow from compound.
+    /// @param amountDai Amount of dai to borrow from maker.
+    function rebalance (uint amountEth, uint amountDai) external onlyRole(STRATEGIST_ROLE) {
```

Comment:
This should use the on-chain price. Price might change.

Reply:
Fixed.